modules/alloy/src/main/ts/ephox/alloy/positioning/mode/NodeAnchor.ts

```diff
@@ -10,15 +11,16 @@ import * as AnchorLayouts from './AnchorLayouts';
 import * as ContainerOffsets from './ContainerOffsets';
 import * as ContentAnchorCommon from './ContentAnchorCommon';
 
-const placement = (component: AlloyComponent, anchorInfo: NodeAnchor, origin: Origins.OriginAdt): Optional<Anchoring> => {
+const placement = (component: AlloyComponent, anchorInfo: NodeAnchor, origin: Origins.OriginAdt, getBounds: Optional<() => Bounds>): Optional<Anchoring> => {
   const rootPoint = ContainerOffsets.getRootPoint(component, origin, anchorInfo);
 
   return anchorInfo.node
     // Ensure the node is still attached, otherwise we can't get a valid client rect for a detached node
     .filter(SugarBody.inBody)
     .bind((target) => {
+      const cappingViewport = Origins.viewport(origin, getBounds);
```

Comment:
There are so many bounds-things everywhere in this code. Given that you are only using `getBounds` to pass it into `Origins.viewport`, would it make it slightly simpler to pass in cappingViewport here rather than getBounds? You'll need to check that the value hasn't changed. It's probably not much difference in readability, anyway.

Reply:
Not sure if it'd make it simpler as such, but in terms of readability that might be a personal opinion. I'd be fine with this, but if people prefer it otherwise I can change it.